scripts/apps/contacts/components/Item.jsx

```diff
@@ -0,0 +1,121 @@
+ */
+export class Item extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {hover: false};
+
+        this.select = this.select.bind(this);
+        this.edit = this.edit.bind(this);
+        this.setHoverState = this.setHoverState.bind(this);
+        this.unsetHoverState = this.unsetHoverState.bind(this);
+    }
+
+    shouldComponentUpdate(nextProps, nextState) {
+        return nextProps.item !== this.props.item ||
+            nextProps.view !== this.props.view ||
+            nextProps.flags.selected !== this.props.flags.selected ||
+            nextState !== this.state;
+    }
+
+    select(event) {
+        this.props.onSelect(this.props.item, event);
+    }
+
+    edit(event) {
+        this.props.onEdit(this.props.item);
+    }
+
+    setHoverState() {
+        this.setState({hover: true});
+    }
+
+    unsetHoverState() {
+        this.setState({hover: false});
+    }
+
+    render() {
+        var item = this.props.item;
+        var contents = [
+            'div',
+            {
+                className: classNames(
+                    'media-box contacts',
+                    {
+                        selected: this.props.flags.selected
+                    }
+                )
+
+            }
+        ];
+
+        if (this.props.view === 'mgrid') {
+            contents.push(
+                React.createElement(ContactHeader, {
+                    item: item,
+                    svc: this.props.svc
+                }),
+                React.createElement(ContactInfo, {
+                    item: item,
+                    svc: this.props.svc
+                }),
+                React.createElement(ContactFooter, {
+                    item: item,
+                    svc: this.props.svc
+                })
+            );
+        } else {
+            contents.push(
+                React.createElement('span', {className: 'state-border'}),
+                React.createElement(ListTypeIcon, {
+                    item: item,
+                    svc: this.props.svc
+                }),
+                React.createElement(ListItemInfo, {
+                    item: item,
+                    svc: this.props.svc,
+                    scope: this.props.scope
+                })
+            );
+        }
+
+        return React.createElement(
+            'li', {
+                id: item._id,
+                key: item._id,
+                className: classNames(
+                    'list-item-view',
+                    {active: this.props.flags.selected}
+                ),
+                onMouseEnter: this.setHoverState,
+                onMouseLeave: this.unsetHoverState,
+                onClick: this.select
+            },
+            React.createElement.apply(null, contents)
+        );
+    }
+}
+
+Item.propTypes = {
+    svc: PropTypes.object.isRequired,
+    scope: PropTypes.any.isRequired,
+    item: PropTypes.any,
+    flags: PropTypes.any,
```

Comment:
Please don't use `any` anywhere. Be _specific_ about the type of the prop. Learn more about this here: https://reactjs.org/docs/typechecking-with-proptypes.html